packages/models/src/actor/registry.ts

```diff
@@ -47,25 +63,46 @@ export class Registry {
     for (const m in exports) {
       const module = exports[m]
       if (typeof module !== 'function') continue
-      this.#bind(module)
+      const metadata = Reflect.getMetadata(ProviderKey.Actor, module)
+      if (metadata?.ref) {
+        this.#router.addPath(metadata.ref, module)
+      }
+      if (metadata?.bindWhenBootstrap) {
+        this.#bind(metadata.ref, module)
+      }
     }
   }
 
   /**
    * this method is defined as public for testing
    */
-  bind = (module: new (...args: Array<unknown>) => unknown): void => this.#bind(module)
+  bind = (module: ConstructorFunction): void => this.#bind(Reflect.getMetadata(ProviderKey.Actor, module)?.ref, module)
```

Comment:
An exception should be thrown if `metadata` is `undefined` because the method `this.#bind` asserts `ref` to be an `ActorReference`